File: generated-sources/scala-finch/mojang-api/src/main/scala/com/github/asyncmc/mojang/api/scala/finch/model/ChangeSkinRequest.scala
package com.github.asyncmc.mojang.api.scala.finch.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import com.github.asyncmc.mojang.api.scala.finch.model.SkinModel

/**
 * Request Mojang to download a skin from an URL and apply to the player
 * @param model 
 * @param url The URL which Mojang servers will download and apply the skin
 */
case class ChangeSkinRequest(model: Option[SkinModel],
                url: String
                )

object ChangeSkinRequest {
    /**
     * Creates the codec for converting ChangeSkinRequest from and to JSON.
     */
    implicit val decoder: Decoder[ChangeSkinRequest] = deriveDecoder
    implicit val encoder: ObjectEncoder[ChangeSkinRequest] = deriveEncoder
}

File: generated-sources/php-laravel/mojang-api/lib/com/github/asyncmc/mojang/api/php/laravel/model/ChangeSkinRequest.php
<?php
/**
 * ChangeSkinRequest
 */
namespace com.github.asyncmc.mojang.api.php.laravel.server\com.github.asyncmc.mojang.api.php.laravel.model;

/**
 * ChangeSkinRequest
 */
class ChangeSkinRequest {

    /** @var \com.github.asyncmc.mojang.api.php.laravel.server\com.github.asyncmc.mojang.api.php.laravel.model\SkinModel $model */
    private $model;

    /** @var string $url The URL which Mojang servers will download and apply the skin*/
    private $url;

}
